v2/rotate.go

```diff
@@ -0,0 +1,30 @@
+package pie
+
+// Rotate return slice circularly rotated by a number of positions n.
+// If n is positive, the slice is rotated right.
+// If n is negative, the slice is rotated left.
+func Rotate[T any](ss []T, n int) []T {
```

Comment:
Please add this to all the Of types so it can also be chained: https://github.com/elliotchance/pie/pull/194/files